proposals/threads/Overview.md

```diff
@@ -297,6 +297,69 @@ returns the number of waiters that were woken as an `i32`.
 
   * `wake`: wake up `wake count` threads waiting on the given address via `i32.wait` or `i64.wait`
 
+## [JavaScript API][] changes
+
+### `WebAssembly.Memory` Constructor
+
+See the current specification [here][WebAssembly.Memory].
+
+The WebAssembly.Memory constructor has the same signature:
+
+```
+new Memory(memoryDescriptor)
+```
+
+However, the `memoryDescriptor` now will check for a `shared` property:
+
+If [HasProperty][]("shared"), then let `shared` be
+[ToBoolean][]([Get][](`memoryDescriptor`, "shared")). Otherwise, let `shared`
+be `false`.
+
+Let `memory` be the result of calling [Memory.create][] given arguments
+`initial`, `maximum`, and `shared`. Note that initial and maximum are specified
+in units of WebAssembly pages (64KiB).
+
+Return the result of [CreateMemoryObject][](`memory`).
+
+### `CreateMemoryObject`
+
+Given a [Memory.memory][] `m`, to create a `WebAssembly.Memory`:
+
+If `m` is shared, let `buffer` be a new [SharedArrayBuffer][]. If `m` is not
+shared, let `buffer` be a new [ArrayBuffer][]. In either case, `buffer` will
+have an internal slot [\[\[ArrayBufferData\]\]][] which aliases `m` and an
+internal slot [\[\[ArrayBufferByteLength\]\]][] which is set to the byte length
+of `m`.
+
+If `m` is shared, any attempts to [detach][] `buffer` shall throw a
+[TypeError][]. Otherwise, any attempts to [detach][] `buffer` _other_ than the
+detachment performed by `m.grow` shall throw a [TypeError][].
```

Comment:
No, I guess this should be clearer. The "Otherwise" is meant to refer to when `m` is not shared; a SAB is never detached.